### ISSUE TITLE
Rustdoc not showing return type but showing return type lifetimes

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rust#52376]
https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir/interpret/struct.Memory.html#method.allocations

![screenshot_2018-07-14 rustc_mir interpret memory - rust](https://user-images.githubusercontent.com/17426603/42725565-f13106c6-8785-11e8-8fd7-61a627bfe688.png)

**Comment by FanShupei** [comment]
It seems the nightly rustdoc cannot handle `impl Trait` well. It works in argument position, not in return position.

I tried with a minimal examle. Create a new lib project, and write following in lib.rs, then cargo doc.

```rust
pub fn f(x: impl Copy) -> impl Copy {
    1
}
```
The output is weird. The return type vanishes.

```
pub fn f(x: impl Copy) -> 
```

rustdoc vesion: rustdoc 1.29.0-nightly (254f8796b 2018-07-13)
toolchain: nightly-x86_64-unknown-linux-gnu

**Comment by GuillaumeGomez** [comment]
Fixed in #53541.
